rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/BakeManifestService.java

```diff
@@ -63,6 +71,25 @@ public BakeManifestService(JobExecutor jobExecutor) {
     if (bakeStatus.getResult() != BakeStatus.Result.SUCCESS) {
       throw new IllegalStateException("Bake failed: " + bakeStatus.getLogsContent());
     }
-    return bakeStatus.getOutputContent().getBytes();
+    return removeTestsDirectoryTemplates(bakeStatus.getOutputContent()).getBytes();
+  }
+
+  public String removeTestsDirectoryTemplates(String input) {
+
+    final List<String> inputManifests =
+        ArrayUtils.toUnmodifiableList(input.split(MANIFEST_SEPARATOR));
+
+    final List<String> outputManifests =
+        inputManifests.stream()
+            .filter(
+                manifest ->
+                    !manifest.trim().isEmpty()
+                        && !Pattern.compile(REGEX_TESTS_MANIFESTS).matcher(manifest).find())
+            .collect(Collectors.toList());
+
+    final StringBuilder stringBuilder = new StringBuilder();
+    outputManifests.stream()
+        .forEach(manifest -> stringBuilder.append(MANIFEST_SEPARATOR).append(manifest));
+    return stringBuilder.toString();
```

Comment:
I think this can be simplified using streams and a joining collector. I haven't tried it but it might be a bit easier to read.
```
String manifestBody = outputManifests.stream().collect(Collectors.joining(MANIFEST_SEPARATOR + "\n"))
return manifestBody.getBytes()
```

Reply:
Fixed